### PR TITLE
update workflows to use new github oidc connection

### PR DESCRIPTION
**Pull request by ppittle**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws-samples/serverless-dotnet-demo/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws-samples/serverless-dotnet-demo/blob/main/CONTRIBUTING.md)
- [ ] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/net-8-pipeline.yaml [hunk 1]
```diff
@@ -49,7 +46,7 @@ jobs:
         - name: configure aws credentials
           uses: aws-actions/configure-aws-credentials@v2
           with:
-            role-to-assume: ${{ secrets.AWS_ROLE }}
+            role-to-assume: ${{ secrets.AWS_LOAD_TEST_RUNNER_ROLE }}
```
Comment (Beau-Gosse-dev) [inline .github/workflows/net-8-pipeline.yaml:49]: Shouldn't this be the deploy role?

Comment (Beau-Gosse-dev) [reply .github/workflows/net-8-pipeline.yaml:49]: same for the other files too

Comment (ppittle) [reply .github/workflows/net-8-pipeline.yaml:53]: went back and re-checked, `AWS_LOAD_TEST_RUNNER_ROLE` is the correct role for the load test jobs.  The Sam Deploy role is used in the Deployment actions.

But did realize I was missing the `permission` statements in some of the docs - I've added those.

File: .github/actions/sam-build-and-deploy-containers/action.yml [hunk 2]
```diff
@@ -36,11 +36,11 @@ runs:
...
       with:
           use-installer: true
-    - uses: aws-actions/configure-aws-credentials@v1
+    - uses: aws-actions/configure-aws-credentials@v2
       with:
-        aws-access-key-id: ${{ inputs.aws-access-key }}
-        aws-secret-access-key: ${{ inputs.aws-secret-key }}
-        aws-region: ${{ inputs.aws-region }}
+        role-to-assume: ${{ secrets.AWS_SAM_DEPLOYMENT_ROLE }}
+        role-session-name: GitHub_to_AWS_via_FederatedOIDC
```
Comment (bhallasaksham) [inline .github/actions/sam-build-and-deploy-containers/action.yml:42]: did we manually create this role in our aws account?

**Pull request merged**